app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb

```diff
@@ -2,9 +2,13 @@ module ManageIQ::Providers::Amazon::CloudManager::Provision::Cloning
   def do_clone_task_check(clone_task_ref)
     source.with_provider_connection(:sdk_v2 => true) do |ec2|
       instance = ec2.instance(clone_task_ref)
-      status   = instance.state.name.to_sym
-      return true if status == :running
-      return false, status
+      if instance.exists?
```

Comment:
```
return false unless instance.exists?

status = instance.state.name.to_sym
return true if status == :running
return false, status
```

would look a little cleaner

Reply:
That does look much cleaner - thanks for the suggestion! I will update the PR.

I still want to return a state, just so the log message here will continue to be useful.

https://github.com/ManageIQ/manageiq/blob/413d9e2c174052435e8c61834f626000fbeff83c/app/models/manageiq/providers/cloud_manager/provision/state_machine.rb#L46

Reply:
:+1: You're right `return false, :pending if yadda yadda`  then